core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -227,10 +229,11 @@ protected void doClose() {
         keepAliveReaper.cancel();
     }
 
-    public DfsSearchResult executeDfsPhase(ShardSearchRequest request) throws IOException {
+    public DfsSearchResult executeDfsPhase(ShardSearchRequest request, SearchTask task) throws IOException {
         final SearchContext context = createAndPutContext(request);
```

Comment:
Maybe `createContext` should take the `SearchTask` as an argument? That'd make it very difficult to forget to set it.